docker/containers.go

```diff
@@ -5,9 +5,31 @@ import (
 
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+func ContainerCreate(
+	config *container.Config,
+	hostConfig *container.HostConfig,
+	networkingConfig *network.NetworkingConfig,
+	platform *specs.Platform,
+	containerName string,
+) (container.CreateResponse, error) {
+	var cli *client.Client
+
+	err := withCli(func(cliInner *client.Client) error {
+		cli = cliInner
+		return nil
+	})
+	if err != nil {
+		return container.CreateResponse{}, err
+	}
+
+	return cli.ContainerCreate(context.Background(), config, hostConfig, networkingConfig, platform, containerName)
```

Comment:
this feels a bit unsafe and beats the purpose of `withCli` fn, isn't cli closed inside `withCli`? is there any reason why not to run these lines of code inside withCli?

Reply:
fixed